packages/autocomplete/src/containers/AutocompleteContainer.js

```diff
@@ -0,0 +1,540 @@
+
+      if (isRtl(this.props)) {
+        if (e.target.value === '' && this.tagSelectionModel.selectedIndex !== 0) {
+          this.tagSelectionModel.selectPrevious();
+          e.preventDefault();
+        }
+      } else {
+        if (
+          e.target.value === '' &&
+          this.tagSelectionModel.selectedIndex === this.tagSelectionModel.numItems - 1
+        ) {
+          this.openDropdown();
+        } else if (!isOpen) {
+          this.tagSelectionModel.selectNext();
+        }
+
+        e.preventDefault();
+      }
+    },
+    [KEY_CODES.HOME]: e => {
+      const { isOpen } = this.getControlledState();
+
+      if (isOpen) {
+        this.focusSelectionModel.selectFirst();
+        e.preventDefault();
+      } else if (e.target.value === '') {
+        this.tagSelectionModel.selectFirst();
+        e.preventDefault();
+      }
+    },
+    [KEY_CODES.END]: e => {
+      const { isOpen } = this.getControlledState();
+
+      if (isOpen) {
+        this.focusSelectionModel.selectLast();
+        e.preventDefault();
+      } else if (e.target.value === '' && this.tagSelectionModel.hasSelection()) {
+        this.openDropdown();
+        e.preventDefault();
+      }
+    },
+    [KEY_CODES.ESCAPE]: e => {
+      const { isOpen } = this.getControlledState();
+
+      if (isOpen) {
+        this.closeDropdown();
+        e.preventDefault();
+      }
+    },
+    [KEY_CODES.ENTER]: e => {
+      const { isOpen } = this.getControlledState();
+
+      if (isOpen) {
+        if (this.focusSelectionModel.hasSelection()) {
+          this.selectItem(this.indexKeyMap[this.focusSelectionModel.selectedIndex]);
+        } else if (Object.keys(this.indexKeyMap).length > 0) {
+          this.selectItem(this.indexKeyMap[0]);
+        }
+
+        e.preventDefault();
+      } else if (!this.tagSelectionModel.hasSelection()) {
+        this.openDropdown();
+        e.preventDefault();
+      }
+    },
+    [KEY_CODES.DOWN]: e => {
+      const { isOpen } = this.getControlledState();
+
+      if (isOpen) {
+        this.focusSelectionModel.selectNext();
+        e.preventDefault();
+      } else {
+        this.focusSelectionModel.selectFirst();
+        e.preventDefault();
+      }
+    },
+    [KEY_CODES.UP]: e => {
+      const { isOpen } = this.getControlledState();
+
+      if (isOpen) {
+        this.focusSelectionModel.selectPrevious();
+        e.preventDefault();
+      } else {
+        this.focusSelectionModel.selectLast();
+        e.preventDefault();
+      }
+    },
+    [KEY_CODES.SPACE]: e => {
+      const { isOpen } = this.getControlledState();
+
+      if (!isOpen) {
+        this.openDropdown();
+        e.preventDefault();
+      }
+    }
+  };
+
+  getItemId = key =>
+    typeof key === 'undefined' ? '' : `${this.getControlledState().id}--item-${key}`;
+
+  getTagId = key =>
+    typeof key === 'undefined' ? '' : `${this.getControlledState().id}--tag-${key}`;
+
+  getMenuId = () => `${this.getControlledState().id}--menu`;
```

Comment:
So I think you had the intention of using this but it's not used in the code base.

Reply:
This is now used now that I'm following the correct W3C design doc.

It's used in the `getInputProps` for `aria-owns` as well as `getMenuProps` for applying the ID